lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +211,50 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Return Doctrine/Mysql-compatible column default values for MariaDB 10.2.7+ servers.
+     *
+     * - Since MariaDb 10.2.7 column defaults stored in information_schema are now quoted
+     *   to distinguish them from expressions (see MDEV-10134 for a what is an expression).
+     * - CURRENT_TIMESTAMP, CURRENT_TIME, CURRENT_DATE are stored in information_schema
+     *   as current_timestamp(), currdate(), currtime()
+     * - Literal escaping is normalized in information schema (store "''" instead of "\'")
+     * - Note: columnDefault should only be null when column is not_nullable otherwise
+     *   a string 'NULL' should be given. Unfortunately, it's not 100% correct in case of
+     *   upgrade from previous versions: see https://jira.mariadb.org/browse/MDEV-14053.
+     *   Otherwise we could simplify the condition === 'NULL' || === null by adding the parameter
+     *   is_nullable.
+     *
+     * @link https://mariadb.com/kb/en/library/information-schema-columns-table/
+     * @link https://jira.mariadb.org/browse/MDEV-13132
+     * @link https://jira.mariadb.org/browse/MDEV-14053
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(MariaDb102Platform $platform, ?string $columnDefault) : ?string {
+
+        if ($columnDefault === 'NULL' || $columnDefault === null) {
```

Comment:
It sucks that we've got to fix the DB's own bugs in the DBAL, especially the upgrade bugs, but given the simplicity of the fix, it should be okay.

Reply:
@morozov , I got updated on this with the maria team and updated the doc. You can follow the bug report here: https://jira.mariadb.org/browse/MDEV-14053.

I finally tracked down the bug and able to reproduce... It happens with views (not so important for doctrine, but also if we dropped the default value: `alter table t1 alter column c drop default;`). 

> From maria: Technically, a table can have a NULLable column without a default value, in which case COLUMN_DEFAULT IS NULL would be true. It usually does not happen, because ways to achieve it are somewhat cumbersome, but nothing specifically prohibits